morango/utils/sync_utils.py

```diff
@@ -26,19 +26,15 @@ def _self_referential_fk(klass_model):
     return None
 
 def _fsic_queuing_calc(fsic1, fsic2):
-    fsic_copy = dict(fsic1)
-    # fsic_copy = dict(fsic2)
-    for k, v in iteritems(fsic1):
-        if k in fsic2:
-            # we remove a (instance id, value) pair if the value is greater in the other fsic
-            if fsic2[k] >= fsic1[k]:
-                del fsic_copy[k]
-            else:
-                fsic_copy[k] = fsic2[k]
-        else:
-            # if instance id is not in other fsic, then zero out value
-            fsic_copy[k] = 0
-    return fsic_copy
+    """
+    We set the lower counter between two same instance ids.
+    If an instance_id exists in one fsic but not the other we want to give that counter a value of 0.
+
+    :param fsic1: dictionary containing (instance_id, counter) pairs
+    :param fsic2: dictionary containing (instance_id, counter) pairs
+    :return ``dict`` of fsics to be used in queueing the correct records to the buffer
+    """
+    return {instance: fsic2.get(instance, -1) for instance, counter in iteritems(fsic1) if fsic2.get(instance, -1) < counter}
```

Comment:
The first counter we set for an instance should be 1, right? So this can just be 0, not -1, can't it? (and that's what the docstring says, too)

Reply:
right, its just tests broke so I opted of changing the logic a bit instead of trying to modify the tests.

Reply:
If the behavior is correct with 0, but it breaks the tests, shouldn't we... fix the tests? When the universe is eventually destroyed, there's a good chance it will be caused by an off-by-one error. :)